server/metrics/metrics.go

```diff
@@ -0,0 +1,147 @@
+package metrics
+
+import (
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+)
+
+const (
+	MetricsNamespace     = "msteams_connect"
+	MetricsSubsystemApp  = "app"
+	MetricsSubsystemHTTP = "http"
+	MetricsSubsystemAPI  = "api"
+
+	MetricsCloudInstallationLabel = "installationId"
+)
+
+type InstanceInfo struct {
+	InstallationID string
+}
+
+// Metrics used to instrumentate metrics in prometheus.
+type Metrics struct {
+	registry *prometheus.Registry
+
+	apiTime *prometheus.HistogramVec
+
+	httpRequestsTotal prometheus.Counter
+	httpErrorsTotal   prometheus.Counter
+
+	connectedUsersTotal prometheus.Gauge
+	syntheticUsersTotal prometheus.Gauge
+	linkedChannelsTotal prometheus.Gauge
+}
+
+// NewMetrics Factory method to create a new metrics collector.
+func NewMetrics(info InstanceInfo) *Metrics {
+	m := &Metrics{}
+
+	m.registry = prometheus.NewRegistry()
+	options := collectors.ProcessCollectorOpts{
+		Namespace: MetricsNamespace,
+	}
+	m.registry.MustRegister(collectors.NewProcessCollector(options))
+	m.registry.MustRegister(collectors.NewGoCollector())
+
+	additionalLabels := map[string]string{}
+	if info.InstallationID != "" {
+		additionalLabels[MetricsCloudInstallationLabel] = os.Getenv("MM_CLOUD_INSTALLATION_ID")
+	}
```

Comment:
Should we actually be using `info.InstallationID` vs. fetching it again via `os.Getenv`?

Reply:
Yes, In theory is exactly the same, but makes a lot of sense. I copied that from playbooks, so if you want I can make a PR for that in playbooks too.

Reply:
Haha, oops! Feel free to push a PR, but also fine to just leave it there -- just good to fix at least this while we're here :)

Reply:
Done